stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py

```diff
@@ -0,0 +1,280 @@
+        operation_lot = self.env['stock.pack.operation.lot']
+        for package in self:
+            picking_type = package.picking_type_id or default_picking_type
+            moves = []
+            for line in package.line_ids:
+                # ----- If line has 'move_id' this means we don't need to
+                #       recreate picking and move again
+                if line.product_id and not line.move_id:
+                    move_data = line.get_move_data()
+                    move_data.update({
+                        'partner_id': package.partner_id.id,
+                        'location_id':
+                            picking_type.default_location_src_id.id,
+                        'location_dest_id':
+                            picking_type.default_location_dest_id.id,
+                        })
+                    moves.append((line, move_data))
+            if moves:
+                picking_data = {
+                    'move_type': 'direct',
+                    'partner_id': package.partner_id.id,
+                    'company_id': package.company_id.id,
+                    'date': package.date,
+                    'picking_type_id': picking_type.id,
+                    'location_id':
+                        picking_type.default_location_src_id.id,
+                    'location_dest_id':
+                        picking_type.default_location_dest_id.id,
+                    }
+                picking = picking_model.create(picking_data)
+                for line, move_data in moves:
+                    move_data.update({'picking_id': picking.id})
+                    move = move_model.create(move_data)
+                    line.move_id = move.id
+                    # ----- Create pack to force lot
+                    if line.lot_id:
+                        operation = pack_model.create({
+                            'product_id': line.product_id.id,
+                            'product_uom_id': line.product_uom.id,
+                            'product_qty': line.product_uom_qty,
+                            'qty_done': line.product_uom_qty,
+                            'location_id': move_data['location_id'],
+                            'location_dest_id': move_data['location_dest_id'],
+                            'date': datetime.now(),
+                            'picking_id': picking.id,
+                            })
+                        operation_lot.create({
+                            'operation_id': operation.id,
+                            'qty': line.product_uom_qty,
+                            'lot_id': line.lot_id.id,
+                        })
+                # ----- Set the picking as "To DO" and try to set it as
+                #       assigned
+                picking.action_confirm()
+                # ----- Show an error if a picking is not confirmed
+                if picking.state != 'confirmed':
+                    raise exceptions.Warning(
```

Comment:
Warning should be renamed as UserError per pylint-odoo. Admittedly don't know the reasoning, but I assume a planned change in the future

```
R8101	Import Warning should be renamed as UserError from openerp.exceptions import Warning as UserError
```